sound/soc/sof/intel/hda-dai.c

```diff
@@ -251,8 +254,10 @@ static int hda_link_dma_trigger(struct snd_pcm_substream *substream, int cmd)
 			stream_tag = hdac_stream(hext_stream)->stream_tag;
 			snd_hdac_ext_link_clear_stream_id(link, stream_tag);
 		}
-
+		snd_soc_dai_set_dma_data(cpu_dai, substream, NULL);
+		snd_hdac_ext_stream_release(hext_stream, HDAC_EXT_STREAM_TYPE_LINK);
```

Comment:
@plbossart is this working without errors for normal playback/capture stop? I think you might need to add if(hext_stream->link_prepared) return 0; in hda_link_dma_hw_free() no?

Reply:
I reverted this change for now, let's try with the code split first to see if I screwed up.